library/core/class.controller.php

```diff
@@ -1270,7 +1278,16 @@ public function xRender($View = '', $ControllerName = false, $ApplicationFolder
             $this->setJson('Data', ($View instanceof Gdn_IModule) ? $View->toString() : $View);
             $this->setJson('InformMessages', $this->_InformMessages);
             $this->setJson('ErrorMessages', $this->_ErrorMessages);
-            $this->setJson('RedirectUrl', $this->RedirectUrl);
+            if ($this->redirectTo !== null) {
+                $this->setJson('RedirectTo', $this->redirectTo);
+                $this->setJson('RedirectUrl', $this->redirectTo);
+            } else if ($this->RedirectUrl !== '') {
+                $this->setJson('RedirectTo', $this->RedirectUrl);
+                $this->setJson('RedirectUrl', $this->RedirectUrl);
+            } else {
+                $this->setJson('RedirectTo', '');
+                $this->setJson('RedirectUrl', '');
```

Comment:
The `RedirectUrl` blocks can probably be simplified into one.

```php
$redirectUrl = $this->RedirectUrl ?: '';
$this->setJson('RedirectTo', $redirectUrl);
$this->setJson('RedirectUrl', $redirectUrl);
```

Unless there a reason we might want to allow any falsy values.